http-svr/jbuild

```diff
@@ -0,0 +1,86 @@
+(* -*- tuareg -*- *)
+#require "unix"
+
+let flags = function
+  | [] -> ""
+  | pkgs ->
+    let cmd = "ocamlfind ocamlc -verbose" ^ (
+        List.fold_left (fun acc pkg -> acc ^ " -package " ^ pkg) "" pkgs
+      ) in
+    let ic = Unix.open_process_in
+        (cmd ^ " | grep -oEe '-ppx (\"([^\"\\]|\\.)+\"|\\w+)'")
+    in
+    let rec go ic acc =
```

Comment:
It probably does not matter but this code is inefficient:

* `go` is not tail recursive
* It would be better to use `Buffer` to build large strings
* Below is a tail-recursive loop to read lines

```let read_lines io =
  let rec loop acc =
      match input_line io with
      | l -> loop (l :: acc)
      | exception End_of_file -> List.rev acc
  in
  loop []
```

Reply:
True but it is usually one or two lines at most, and it's the same boilerplate on all our jbuild files. I plan to replace it with `ppx_find` and remove it completely so I don't think it is worth touching